### PR TITLE
Add %rotate, fix defines and macro match

### PR DESCRIPTION
**Pull request by Super-Pizza**
Fixes the following matches
  - `%rotate`
  - `%idefine`,`%xdefine`,`%ixdefine`,`%define`
  - `%imacro`

### CODE REVIEW

File: x86_64 Assembly.tmbundle/Syntaxes/Nasm Assembly.sublime-syntax [hunk 1]
```diff
@@ -544,7 +544,7 @@ contexts:
         - include: strings
         - include: line-ending
         - include: preprocessor-macro-indirection
-    - match: '^\s*((%)(?:assign|i?deftok|strcat|strlen|substr|pathsearch|push|pop|repl|line|clear))\b'
+    - match: '^\s*((%)(?:assign|i?deftok|strcat|strlen|substr|pathsearch|push|pop|rotate|repl|line|clear))\b'
```
Comment (13xforever) [inline x86_64 Assembly.tmbundle/Syntaxes/Nasm Assembly.sublime-syntax:547]: `%rotate` is only defined inside a multi-line macro, and it is already covered (see line 863)

File: x86_64 Assembly.tmbundle/Syntaxes/x86_64 Assembly.YAML-tmLanguage [hunk 2]
```diff
@@ -188,7 +188,7 @@ repository:
           '0': {name: punctuation.definition.string.end.c}
     
     - name: meta.preprocessor.c
-      begin: ^\s*[%#]\s*((xi?)?define|defined|elif(def)?|else|if(macro|ctx|idni?|num|str)?|ifn?def|line|(end)?macro|pragma|undef|endif)\b
+      begin: ^\s*[%#]\s*(i?x?define|defined|elif(def)?|else|if(macro|ctx|idni?|num|str)?|ifn?def|line|(end|i)?macro|pragma|undef|endif)\b
```
Comment (13xforever) [inline x86_64 Assembly.tmbundle/Syntaxes/x86_64 Assembly.YAML-tmLanguage:191]: I'd prefer the logical order for macro matching, e.g. `(i|end|un)?macro`

**Comment by 13xforever** [comment]
`.sublime-syntax` is built from scratch up-to-spec, and will be used for st3 and newer versions
`.YAML-tmLanguage` is an old version for st2 and older (and also for other editors such as vscode), and that one has major issues and needs a rewrite as well (but it's much more limited compared to `.sublime-syntax`)

there's a couple small changes I'd like to see before merging

**Comment by Super-Pizza** [comment]
I applied your requests, what else would you like to change?

**Pull request merged**